aiida/transports/plugins/ssh.py

```diff
@@ -1228,7 +1256,7 @@ def _exec_command_internal(self, command, combine_stderr=False, bufsize=-1):  #
         else:
             command_to_execute = command
 
-        self.logger.debug('Command to be executed: {}'.format(command_to_execute))
+        self.logger.debug('Command to be executed: {}'.format(command_to_execute[:1000]))
```

Comment:
So the rest of the re-organization and wrapping of methods seems ok to me, but this is the only modification that I am unsure about. Can't you be loosing important log info like this? I would also maybe ask for the input of @giovannipizzi and/or @sphuber in this, since they probably know better what to expect inside that command and why is being logged.

Reply:
Also interested to know why @adegomme thinks this should be cropped.

Reply:
To send a file with the sshonly plugin, the resulting command is equivalent to `ssh remote  "echo '$(cat localfile)' | cat > remotefile" ` , with the '$(cat localfile)' being a python read instead, as I never managed to have the cat version behave correctly.
This logger call means that the whole content of the file is printed in the log :
example when making a 'verdi computer test'
09/17/2020 10:17:12 AM <1120861> aiida.transport.SshOnlyTransport: [DEBUG] Command to be executed: cd '/path/aiida' && echo 'Test from '"'"'verdi computer test'"'"' on 2020-09-17T10:17:05.830707' | cat > /path/aiida/tmpbtteknws

So this adds the whole file content to the log output when debug is activated, which is not great. Every input file, structure file, pseudopotential file gets printed in the stdout or the log when debug is on, and when several MB get dumped on you, this can be troublesome.
Cropping is meant to avoid ruining the log and the terminal. 1000 char is a bit arbitrary, and maybe we could add a comment when it's cropped ( simply "...", or "cropped to 1000 chars to maintain readability") to avoid confusion. 
Another idea would be to have an optional parameter on _exec_command_internal  to disable this log message, but you would not see any file upload log in this case.

Reply:
I see. Thanks for the explanation. In any case, even if we keep this, we should definitely comment this, because it is not obvious at all, especially given that the actual use case comes from a subclass that lives in another plugin package. @giovannipizzi do you see a way around this? If we make the cropping size an optional argument to the function, at least `SshOnly` can override the copy file method to call `_exec_command_internal` to pass `max_log_size=1000` or something to that effect.

Reply:
After discussing with @sphuber I think a good solution is to have a class attribute, e.g. `_MAX_EXEC_COMMAND_LOG_SIZE=None` in this base class.
Here the logic would not crop if `self._MAX_EXEC_COMMAND_LOG_SIZE` is None, otherwise it would crop to this length (`command_to_execute[:self._MAX_EXEC_COMMAND_LOG_SIZE]`).

In this way the current behaviour is unchanged, and in your subclass @adegomme  you can just set `_MAX_EXEC_COMMAND_LOG_SIZE` to the value you want.

@adegomme what do you think? Are you OK to do this small change? After, it's ready to merge for me